mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/kafka/Consumer.java

```diff
@@ -1,16 +1,103 @@
 package de.brandwatch.minianalytics.mentiongenerator.kafka;
 
+import de.brandwatch.minianalytics.mentiongenerator.model.Mention;
+import de.brandwatch.minianalytics.mentiongenerator.model.Resource;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.model.Query;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.repository.QueryRepository;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.RAMDirectory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.kafka.annotation.KafkaListener;
 
+import java.io.IOException;
+import java.util.List;
+
 public class Consumer {
 
     private static final Logger logger = LoggerFactory.getLogger(Consumer.class);
 
+    private QueryRepository queryRepository;
+
+    private Producer producer;
+
+    private Directory memoryIndex = new RAMDirectory();
+    private StandardAnalyzer analyzer = new StandardAnalyzer();
+
+    private IndexWriterConfig indexWriterConfig = new IndexWriterConfig(analyzer);
+    private IndexWriter writer = new IndexWriter(memoryIndex, indexWriterConfig);
+
+    private IndexSearcher searcher;
+    private IndexReader reader;
+
+    private QueryParser queryParser = new QueryParser("text", analyzer);
+
+    @Autowired
+    public Consumer(QueryRepository queryRepository, Producer producer) throws IOException {
+        this.queryRepository = queryRepository;
+        this.producer = producer;
+    }
+
     @KafkaListener(topics = "twitter")
```

Comment:
Shouldn't these be on a `resources` topic?

Reply:
There is no `resources` topic. I just named it `twitter`. But I can rename it.

Reply:
It shouldn't be called twitter as it has by design nothing to do with twitter 🤷‍♂

Reply:
Changed to `resources`